subprojects/docs/src/docs/release/notes.md

```diff
@@ -181,6 +181,37 @@ This allows configuration of PMD to run its analysis on more than one thread.
 
 See the [documentation](userguide/pmd_plugin.html#sec:pmd_conf_threads) for more information.
 
+### Configuration cache improvements
+
+#### Modification of system properties at the configuration time
+
+Changes to system properties made at the configuration time are now also applied when the configuration cache entry is reused.
+This includes adding a property, modifying or removing an existing one.
+The types used for property keys and values have to be [supported by the configuration cache](userguide/configuration_cache.html#config_cache:requirements:disallowed_types).
+
+#### Running external processes at the configuration time
+
+Running external processes at the configuration time with Java or Groovy APIs or with `exec` and `javaexec` Gradle APIs is
+now [disallowed](userguide/configuration_cache.html#config_cache:requirements:external_processes) if the configuration cache is enabled.
+A [new Provider-based API](javadoc/org/gradle/api/provider/ProviderFactory.html#exec-org.gradle.api.Action-) is
+now available to obtain an output of the external process in the configuration-cache-compatible way.
+If a more complex interaction with the external process is necessary, then a custom [`ValueSource`](javadoc/org/gradle/api/provider/ValueSource.html) implementation
+can be used. The injected [`ExecOperations`](javadoc/org/gradle/process/ExecOperations.html) service should be used to run the external process.
+
+#### Accessing the environment in the ValueSource implementation no longer introduces additional build configuration inputs
+
+Initially, the [`ValueSource`](javadoc/org/gradle/api/provider/ValueSource.html) implementations were subject to the same input tracking rules as the rest of the build logic.
+Reading a file, an environment variable, or a system property would make this a build configuration input.
+Any change to the input would invalidate the configuration cache, even if the value of the `ValueSource` doesn't change.
+
+Now, reading a file, an environment variable, or a system property is no longer tracked inside the implementation of the `ValueSource`.
+The value of the `ValueSource` is still recomputed each time the build runs, and the configuration cache entry is only invalidated if the value changes.
+
+#### Files read at the configuration time become build configuration inputs
+
+Files read at the configuration time with `FileInputStream` or some Kotlin APIs now become build configuration inputs.
```

Comment:
We should say that before we ignored them and that was a correctness issue.

Reply:
Done